homeassistant/components/isy994.py

```diff
@@ -6,213 +6,274 @@
 """
 import logging
 from urllib.parse import urlparse
+import voluptuous as vol
 
+from homeassistant.core import HomeAssistant  # noqa
 from homeassistant.const import (
     CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
     EVENT_HOMEASSISTANT_STOP)
-from homeassistant.helpers import validate_config, discovery
-from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers import discovery, config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType, Dict  # noqa
+
 
 DOMAIN = "isy994"
-REQUIREMENTS = ['PyISY==1.0.6']
+REQUIREMENTS = ['PyISY==1.0.7']
 
 ISY = None
-SENSOR_STRING = 'Sensor'
-HIDDEN_STRING = '{HIDE ME}'
+DEFAULT_SENSOR_STRING = 'sensor'
+DEFAULT_HIDDEN_STRING = '{HIDE ME}'
 CONF_TLS_VER = 'tls'
+CONF_HIDDEN_STRING = 'hidden_string'
+CONF_SENSOR_STRING = 'sensor_string'
+KEY_MY_PROGRAMS = 'My Programs'
+KEY_FOLDER = 'folder'
+KEY_ACTIONS = 'actions'
+KEY_STATUS = 'status'
 
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.url,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_TLS_VER, None): vol.Coerce(float),
+        vol.Optional(CONF_HIDDEN_STRING, DEFAULT_HIDDEN_STRING): cv.string,
+        vol.Optional(CONF_SENSOR_STRING, DEFAULT_SENSOR_STRING): cv.string
+    })
```

Comment:
lol, this would so benefit from a string schema! #3123 ftw

Your defaults might be wrong:
```python
- vol.Optional(CONF_HIDDEN_STRING, DEFAULT_HIDDEN_STRING): cv.string
+ vol.Optional(CONF_HIDDEN_STRING, default=DEFAULT_HIDDEN_STRING): cv.string
```